restructured as a package (updated)

nobodynate opened the pull request:
Previous PR was from the wrong branch.
#### Overview of Changes
- added pyproject
- restructured as a package

#### Testing
- successfully installed on WSL and MacOS via pipx
- configuration works
- created and destroyed the ec2_ssrf scenario

TeneBrae93 reviewed (dismissed):
This all looks good to me. I reviewed the code & was able to install it with pipx on Kali with no issues. I also tested launching and destroying my sns_secrets scenario and everything worked well.

nobodynate commented:
Updated the dockerfile
- tested build and verified `cloudgoat` works in the container.
Updated Readme
- match/replaced "./cloudgoat.py" to "cloudgoat"
- replaced the old quick start install with pipx install

TeneBrae93 reviewed (approved):
Looks good to me.
